plugins/API/API.php

```diff
@@ -838,6 +839,15 @@ public function getStylesheetFiles(&$stylesheets)
         $stylesheets[] = "plugins/API/stylesheets/glossary.less";
     }
 
+    public function getJsGlobalVariables(&$out)
```

Comment:
I know this probably follows an already established pattern of method naming etc, but can we start adding some, at least simple method docblocks?

Just quickly checked and e.g. the [LanguagesManager](https://github.com/matomo-org/matomo/blob/e66c638ae145e52b2de46c46aae49eb1a0970df4/plugins/LanguagesManager/LanguagesManager.php#L86) does have some more info there.

Just a suggestion. Seeing the method bare like this always reminds me of the journey ahead of us. I think even the PhpStorm generated default with the param name and type would be better than nothing from my perspective.

Apologies for a slightly unrelated topic, it's ok to leave it as is, but I needed to express my view here.

Reply:
Yes. It's currently hard to see which methods defined in a plugin are used for a certain event or not. That is defined in the `registerEvents` method, though. But it can easily be overseen if an event is removed there, but the method remains unused.

I would refactor that system in a way, that the `registerEvents` methods is obsolete. Instead every method used for an event would have a doc bloc with a comment like `@matomoEvent Platform.initialized`. 
We would then need some additional functionality that uses reflection to gather such methods and register them automatically. Some sort of caching for this would also be needed, to avoid using reflection on each call.
That way would force a doc bloc and make it much easier to understand imho.